### PR TITLE
Smartcab - Reward issues fixed

### PR DESCRIPTION
**Pull request by consideRatio**
The latest commit [smartcab left turn](https://github.com/udacity/machine-learning/commit/912f9edbb2c6647039cf71c42f0f075d842b273e) left some issues unresolved. The commit made you crash if you tried turning left when there were oncoming traffic turning right or going forward. The issue remaining was that waiting for such oncoming traffic led to a minor violation, and no bonus in reward.

UPDATE: The [theoretically optimal policy](https://github.com/udacity/machine-learning/files/1275514/optimal_policy.txt)* will still get an A+ rating in reliability with the current default traffic intensity and deadline.

\* This theoretically optimal policy is based on traffic rules and normal behavior, such as waiting on oncoming traffic when about to turn left instead of making three right turns or going forward instead.

### CODE REVIEW

**Comment by consideRatio** [comment]
Ping @adarsh0806

File: projects/smartcab/smartcab/environment.py [hunk 1]
```diff
@@ -327,21 +327,31 @@ def act(self, agent, action):
...
+                if (agent.get_next_waypoint() == 'left' and (inputs['oncoming'] == 'right' or inputs['oncoming'] == 'forward')):
+                    pass
+                else:
+                    violation = 1 # Minor violation
 
 
         # Did the agent attempt a valid move?
         if violation == 0:
-            if action == agent.get_next_waypoint(): # Was it the correct action?
+            # Agent moved wihtout violations or accidents towards the waypoint.
```
Comment (giladgressel) [inline projects/smartcab/smartcab/environment.py]: typo on the word "without"

File: projects/smartcab/smartcab/environment.py [hunk 2]
```diff
@@ -327,21 +327,31 @@ def act(self, agent, action):
...
+            if action == agent.get_next_waypoint():
                 reward += 2 - penalty # (2, 1)
-            elif action == None and light != 'green' and agent.get_next_waypoint() == 'right':
-                # valid action but incorrect (idling at red light, when we should have gone right on red)
-                reward += 1 - penalty # (1, 0)
-            elif action == None and light != 'green': # Was the agent stuck at a red light?
-                reward += 2 - penalty # (2, 1)
-            else: # Valid but incorrect
+            # Agent moved wihtout violations or accidents, but not towards the waypoint.
```
Comment (giladgressel) [inline projects/smartcab/smartcab/environment.py]: same typo here

**Comment by giladgressel** [comment]
this change seems good to me.  In summary you are adding rewards for idling at green lights correctly.

I would have wanted to see some kind of update where we can actually make safe left turns (by yielding) but that is too complex for this environment.

**Comment by consideRatio** [comment]
Thank you for looking through the PR @giladgressel !!

**Pull request closed without merging**